controllers/article_controller.js

```diff
@@ -18,9 +19,14 @@ exports.getArticleById = (req, res, next) => {
 };
 
 exports.getAllArticles = (req, res, next) => {
-  selectAllArticles().then((article) => {
-    res.status(200).send(article);
-  });
+  const { topic } = req.query;
+  selectAllArticles(topic)
+    .then((article) => {
+      res.status(200).send({ article });
+    })
+    .catch((err) => {
+      next(err);
+    });
```

Comment:
one way we could do the 'paper' test, is by having if logic here - if there's a topic, do the Promise.all, if not, then do the regular selectAllArticles. It wouldn't be ideal, but it should work :)

Reply:
Thank you, I shall have a look after finishing all the CORE tickets.